core/listings.go

```diff
@@ -140,86 +98,104 @@ func (n *OpenBazaarNode) UpdateListing(r []byte, publish bool) error {
 	return n.saveListing(listing, publish)
 }
 
-func prepListingForPublish(n *OpenBazaarNode, listing repo.Listing) error {
-	mods, err := listing.GetModerators()
-	if err != nil {
+func (n *OpenBazaarNode) validateListingIsSellable(l repo.Listing) error {
+	var isTestnet = n.TestNetworkEnabled() || n.RegressionNetworkEnabled()
+
+	if err := l.ValidateListing(isTestnet); err != nil {
 		return err
 	}
+
+	var acceptedCurrenciesSeen = make(map[string]bool)
+	for _, c := range l.GetAcceptedCurrencies() {
+		if _, err := n.Multiwallet.WalletForCurrencyCode(c); err != nil {
+			return fmt.Errorf("currency (%s) not supported by wallet", c)
+		}
+
+		currDef, err := repo.AllCurrencies().Lookup(c)
+		if err != nil {
+			return fmt.Errorf("lookup currency (%s): %s", c, err.Error())
+		}
+
+		if acceptedCurrenciesSeen[currDef.CurrencyCode().String()] {
+			return errors.New("duplicate accepted currency in listing")
+		}
+		acceptedCurrenciesSeen[currDef.CurrencyCode().String()] = true
+	}
+
+	return nil
+}
+
+func (n *OpenBazaarNode) saveListing(l repo.Listing, publish bool) error {
+	mods := l.GetModerators()
 	if len(mods) == 0 {
 		sd, err := n.Datastore.Settings().Get()
 		if err == nil && sd.StoreModerators != nil {
-			err = listing.SetModerators(*sd.StoreModerators)
-			if err != nil {
+			if err := l.SetModerators(*sd.StoreModerators); err != nil {
 				return err
 			}
 		}
 	}
 
-	ct, err := listing.GetContractType()
-	if err != nil {
-		return err
-	}
+	ct := l.GetContractType()
 	if pb.Listing_Metadata_ContractType_value[ct] == int32(pb.Listing_Metadata_CRYPTOCURRENCY) {
-		err = listing.ValidateCryptoListing()
-		if err != nil {
+		if err := l.ValidateCryptoListing(); err != nil {
 			return err
 		}
 
-		err = listing.SetCryptocurrencyListingDefaults()
-		if err != nil {
+		if err := l.SetCryptocurrencyListingDefaults(); err != nil {
 			return err
 		}
 	}
 
-	err = n.SetListingInventory(listing)
-	if err != nil {
+	if err := n.validateListingIsSellable(l); err != nil {
+		return fmt.Errorf("validate sellable listing (%s): %s", l.GetSlug(), err.Error())
```

Comment:
This line is what's causing the API test to fail. Basically this used to return a type `repo.ErrPriceModifierOutOfRange` but this changes it to return a generic error. 

We can either modify the API test to just check the string representation of the expected error or we need to create a `ErrSellableListingError`